openssl-dynamic/src/main/c/sslutils.c

```diff
@@ -565,39 +570,30 @@ int tcn_X509_up_ref(X509* cert) {
 #endif
 }
 
-/*
- * This OpenSSL callback function is called when OpenSSL
- * does client authentication and verifies the certificate chain.
- */
-int SSL_callback_SSL_verify(int ok, X509_STORE_CTX *ctx)
-{
-   /* Get Apache context back through OpenSSL context */
-    SSL *ssl = X509_STORE_CTX_get_ex_data(ctx,
-                                          SSL_get_ex_data_X509_STORE_CTX_idx());
-    tcn_ssl_ctxt_t *c = SSL_get_app_data2(ssl);
-
-    /* Get verify ingredients */
-    int errnum   = X509_STORE_CTX_get_error(ctx);
-    int errdepth = X509_STORE_CTX_get_error_depth(ctx);
-    int verify   = c->verify_mode;
-    int depth    = c->verify_depth;
-
-    if (verify == SSL_CVERIFY_UNSET ||
-        verify == SSL_CVERIFY_NONE)
-        return 1;
-
-    if (SSL_VERIFY_ERROR_IS_OPTIONAL(errnum) &&
-        (verify == SSL_CVERIFY_OPTIONAL_NO_CA)) {
-        ok = 1;
+int tcn_set_verify_config(tcn_ssl_verify_config_t* c, jint tcn_mode, jint depth) {
+    if (depth > 0) {
+        c->verify_depth = depth;
+    } else {
+        c->verify_depth = 0;
     }
 
-    if (errdepth > depth) {
-        /* TODO: Some logging
-         * Certificate Verification: Certificate Chain too long
-         */
-        ok = 0;
+    switch (tcn_mode) {
+      case SSL_CVERIFY_IGNORED:
+        switch (c->verify_mode) {
+          case SSL_CVERIFY_NONE: return SSL_VERIFY_NONE;
```

Comment:
nit: not strong on this but I think its more readable to have the return in the next line